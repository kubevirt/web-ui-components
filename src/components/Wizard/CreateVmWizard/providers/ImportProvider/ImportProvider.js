// eslint-disable-next-line no-unused-vars
import React from 'react';

import PropTypes from 'prop-types';

export const ImportProvider = ({ children, isVisible }) => (isVisible ? children : null);

ImportProvider.displayName = 'ImportProvider';

ImportProvider.defaultProps = {
  isVisible: false,
  children: null,
};

ImportProvider.propTypes = {
  isVisible: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};
