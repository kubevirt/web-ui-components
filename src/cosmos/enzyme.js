import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
