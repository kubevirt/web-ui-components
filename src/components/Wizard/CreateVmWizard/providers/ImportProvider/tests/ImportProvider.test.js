import React from 'react';
import { mount } from 'enzyme';

import { ImportProvider } from '../ImportProvider';

const testImportProvider = () => <ImportProvider isVisible>provider</ImportProvider>;

describe('<ImportProvider />', () => {
  it('renders correctly', () => {
    const component = mount(testImportProvider());
    expect(component).toMatchSnapshot();
  });
});
