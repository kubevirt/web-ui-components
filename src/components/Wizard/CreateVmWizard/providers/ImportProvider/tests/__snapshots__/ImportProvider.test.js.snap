// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportProvider /> renders correctly 1`] = `
<ImportProvider
  isVisible={true}
>
  provider
</ImportProvider>
`;
